rocAL/rocAL/source/meta_data/meta_node_rotate.cpp

```diff
@@ -67,25 +62,26 @@ void RotateMetaNode::update_parameters(pMetaDataBatch input_meta_data, pMetaData
             rotate[2] = -1 * rotate[1];
             dest_cx = _dst_width / 2;
             dest_cy = _dst_height / 2;
-            src_cx = _src_width_val[i]/2;
-            src_cy = _src_height_val[i]/2;
+
+            src_cx = static_cast<float>(input_roi[i].x2) / 2;
+            src_cy = static_cast<float>(input_roi[i].y2) / 2;
             src_bb_x = coords_buf[j].l;
             src_bb_y = coords_buf[j].t;
-            bb_w = coords_buf[j].r;
-            bb_h = coords_buf[j].b;
+            bb_w = coords_buf[j].r - coords_buf[j].l;
+            bb_h = coords_buf[j].b - coords_buf[j].t;
             x1 = (rotate[0] * (src_bb_x - src_cx)) + (rotate[1] * (src_bb_y - src_cy)) + dest_cx;
             y1 = (rotate[2] * (src_bb_x - src_cx)) + (rotate[3] * (src_bb_y - src_cy)) + dest_cy;
-            x2 = (rotate[0] * ((src_bb_x + bb_w) - src_cx))+( rotate[1] * (src_bb_y - src_cy)) + dest_cx;
-            y2 = (rotate[2] * ((src_bb_x + bb_w) - src_cx))+( rotate[3] * (src_bb_y - src_cy)) + dest_cy;
+            x2 = (rotate[0] * ((src_bb_x + bb_w) - src_cx)) + ( rotate[1] * (src_bb_y - src_cy)) + dest_cx;
+            y2 = (rotate[2] * ((src_bb_x + bb_w) - src_cx)) + ( rotate[3] * (src_bb_y - src_cy)) + dest_cy;
             x3 = (rotate[0] * (src_bb_x - src_cx)) + (rotate[1] * ((src_bb_y + bb_h) - src_cy)) + dest_cx;
             y3 = (rotate[2] * (src_bb_x - src_cx)) + (rotate[3] * ((src_bb_y + bb_h) - src_cy)) + dest_cy;
-            x4 = (rotate[0] * ((src_bb_x + bb_w) - src_cx))+( rotate[1] * ((src_bb_y + bb_h) - src_cy)) + dest_cx;
-            y4 = (rotate[2] * ((src_bb_x + bb_w) - src_cx))+( rotate[3] * ((src_bb_y + bb_h) - src_cy)) + dest_cy;
+            x4 = (rotate[0] * ((src_bb_x + bb_w) - src_cx)) + ( rotate[1] * ((src_bb_y + bb_h) - src_cy)) + dest_cx;
+            y4 = (rotate[2] * ((src_bb_x + bb_w) - src_cx)) + ( rotate[3] * ((src_bb_y + bb_h) - src_cy)) + dest_cy;
             min_x = std::min(x1, std::min(x2, std::min(x3, x4)));
             min_y = std::min(y1, std::min(y2, std::min(y3, y4)));
-            box.l = std::max(min_x, 0.0f);
-            box.t = std::max(min_y, 0.0f);
-            box.r = std::max(x1, std::max(x2, std::max(x3, x4))); ;
+            box.l = std::min(0.0f, min_x);
```

Comment:
is this std::min or std::max ? Please double check. Also we need to rerun all related trainings for code coverage since a lot of code has been changed for bbox computations

Reply:
We checked the metanode outputs of rotate by printing bboxes on rotated images. Its not working in ToT so we made changes to the algorithm to make it work in tensor branch.
For code coverage, we have ran mask and SSD trainings with the modified tensor branch to confirm the convergence and accuracy.